src/OrleansRuntime/MembershipService/MembershipTableFactory.cs

```diff
@@ -22,50 +35,73 @@ public MembershipTableFactory(IServiceProvider serviceProvider, LoggerWrapper<Me
             this.logger = logger;
         }
 
+        private async Task<IMembershipTable> GetMembershipTableLegacy(GlobalConfiguration globalConfiguration)
```

Comment:
IMO this should be done a little bit differently. Basically configure with the legacy configuration in DI as soon as you pass in a legacy configuration to load.
By the time we ask for the membership table at runtime there should be no fallback, and this should have already been configured.
I can submit a PR to your branch if it's not clear

Reply:
The challenge there is you don't know the user configure with legacy way or the new way. And the IServiceCollection interface doesn't provide you much flexibility to help there.
I'm about to apply the same to Gatewaylist provider. let's discuss what you mean here